File: frontend/src/types/price.ts
export interface PriceEntry {
  condition: string
  variant: string
  low_price: number | null
  mid_price: number | null
  high_price: number | null
  market_price: number | null
}

export interface MarketplacePrices {
  marketplace: string
  url: string | null
  prices: PriceEntry[]
}

export interface PriceComparison {
  card_id: number
  card_name: string
  fetched_at: string
  marketplaces: MarketplacePrices[]
}

export interface PriceHistoryPoint {
  date: string
  marketplace: string
  variant: string
  market_price: number
}

export interface ChartDataPoint {
  date: string
  tcgplayer?: number
  ebay?: number
}

File: backend/app/schemas/price.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class PriceEntry(BaseModel):
    condition: str
    variant: str
    low_price: Optional[float] = None
    mid_price: Optional[float] = None
    high_price: Optional[float] = None
    market_price: Optional[float] = None

    model_config = {"from_attributes": True}


class MarketplacePrices(BaseModel):
    marketplace: str
    url: Optional[str] = None
    prices: list[PriceEntry]


class PriceComparisonResponse(BaseModel):
    card_id: int
    card_name: str
    fetched_at: datetime
    marketplaces: list[MarketplacePrices]


class PriceHistoryPoint(BaseModel):
    date: datetime
    marketplace: str
    variant: str
    market_price: float

    model_config = {"from_attributes": True}
